## 📋 Quality Gate Results

| Check | Status |
|-------|--------|
| Knip | ✅ success |
| Lint | ❌ failure |
| Build | ❌ failure |
| Infra Tests | ❌ skipped |
| Unit Tests | ❌ skipped |
| Component Tests | ❌ skipped |
| Perf Tests | ❌ skipped |
| Visual Tests | ❌ skipped |


### ❌ Lint Failure Details
```text

> hrm@0.30.0 lint /home/runner/work/hrm/hrm
> eslint app/ components/ constants/ context/ hooks/ lib/ services/ tests/ types/ utils/ server.ts proxy.ts --cache


/home/runner/work/hrm/hrm/app/api/spotify/control/route.ts
   21:12  warning  '_e' is defined but never used  unused-imports/no-unused-vars
  123:16  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/hooks/useBluetoothHRM.test.ts
  161:16  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/hooks/useBluetoothHRM.ts
  519:18  warning  '_err' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/hooks/usePersistentStorage.ts
  14:12  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/lib/spotify.ts
  56:14  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/services/spotifyPolling.ts
  126:13  warning  'refresh_token' is assigned a value but never used. Allowed unused vars must match /^_/u  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/tests/integration/test-helpers.ts
  70:18  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/tests/playwright/infrastructure.spec.ts
   98:16  warning  '_e' is defined but never used  unused-imports/no-unused-vars
  140:16  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/tests/playwright/lib/visual.ts
  89:12  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/tests/playwright/performance.spec.ts
  57:16  warning  '_error' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/tests/playwright/timer-ui-sync.spec.ts
  50:14  warning  '_e' is defined but never used  unused-imports/no-unused-vars

/home/runner/work/hrm/hrm/tests/unit/socketManager.test.ts
  439:11  error  'mockBroadcast' is never reassigned. Use 'const' instead  prefer-const

✖ 14 problems (1 error, 13 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

 ELIFECYCLE  Command failed with exit code 1.
```
### ❌ Build Failure Details
```text

> hrm@0.30.0 prebuild /home/runner/work/hrm/hrm
> pnpm run clean


> hrm@0.30.0 clean /home/runner/work/hrm/hrm
> rm -rf .next dist coverage test-results playwright-report *.log *.backup nohup.out diff.txt


> hrm@0.30.0 build /home/runner/work/hrm/hrm
> pnpm run build:server && next build --webpack


> hrm@0.30.0 build:server /home/runner/work/hrm/hrm
> pnpm exec tsc -p tsconfig.build.json

utils/socketManager.ts(317,13): error TS2353: Object literal may only specify known properties, and 'timestamp' does not exist in type 'HrmStreamData'.
 ELIFECYCLE  Command failed with exit code 2.
 ELIFECYCLE  Command failed with exit code 2.
```
### ❌ Infrastructure Test Failure Details
```text
Log file not found.
```
### ❌ Unit Test Failure Details
```text
Log file not found.
```
### ❌ Component Test Failure Details
```text
Log file not found.
```
### ❌ Visual Test Failure Details
```text
Log file not found.
```
### ❌ Performance Test Failure Details
```text
Log file not found.
```
⚠️ **Some checks failed.** Full logs available in [workflow artifacts](https://github.com/arii/hrm/actions/runs/21551422861).

---
> Report generated for commit: `acc6677474124ddca3bd3364ada0ac54a02fcf4d`

--- a/tests/unit/socketManager.test.ts
+++ b/tests/unit/socketManager.test.ts
@@ -352,7 +352,7 @@ describe('WebSocket Manager', () => {
       })
       mockWs.emit('message', baselineMessage.toString())
 
-      let mockBroadcast = broadcast as jest.Mock
+      const mockBroadcast = broadcast as jest.Mock
       let lastCall =
         mockBroadcast.mock.calls[mockBroadcast.mock.calls.length - 1]
       let finalPayload: HrmData[] = lastCall[1].payload
@@ -436,7 +436,7 @@ describe('WebSocket Manager', () => {
         })
       )
 
-      let mockBroadcast = broadcast as jest.Mock
+      const mockBroadcast = broadcast as jest.Mock
       let lastCall =
         mockBroadcast.mock.calls[mockBroadcast.mock.calls.length - 1]
       let finalPayload: HrmData[] = lastCall[1].payload

--- a/tests/unit/socketManager.test.ts
+++ b/tests/unit/socketManager.test.ts
@@ -352,7 +352,7 @@ describe('WebSocket Manager', () => {
       })
       mockWs.emit('message', baselineMessage.toString())
 
-      const mockBroadcast = broadcast as jest.Mock
+      let mockBroadcast = broadcast as jest.Mock
       let lastCall =
         mockBroadcast.mock.calls[mockBroadcast.mock.calls.length - 1]
       let finalPayload: HrmData[] = lastCall[1].payload

--- a/types/core.ts
+++ b/types/core.ts
@@ -95,6 +95,7 @@ export interface HrmStreamData {
   age?: number
   calories: number
   weightKg?: number
+  timestamp?: number
 }
 
 /**

